deepcell_toolbox/multiplex_utils.py

```diff
@@ -0,0 +1,151 @@
+            each cellular compartment with a model prediction. Each key maps to a subsequent dict
+            with the following keys entries
+            - inner-distance: Prediction for the inner distance transform.
+            - outer-distance: Prediction for the outer distance transform
+            - fgbg-fg: prediction for the foreground/background transform
+            - pixelwise-interior: Prediction for the interior/border/background transform.
+        compartment: which cellular compartments to generate predictions for.
+            must be one of 'whole_cell', 'nuclear', 'both'
+        whole_cell_kwargs (dict): Optional list of post-processing kwargs for whole-cell prediction
+        nuclear_kwargs (dict): Optional list of post-processing kwargs for nuclear prediction
+
+    Returns:
+        numpy.array: Uniquely labeled mask for each compartment
+
+    Raises:
+        ValueError: for invalid compartment flag
+    """
+
+    valid_compartments = ['whole-cell', 'nuclear', 'both']
+
+    if whole_cell_kwargs is None:
+        whole_cell_kwargs = {}
+
+    if nuclear_kwargs is None:
+        nuclear_kwargs = {}
+
+    if compartment not in valid_compartments:
+        raise ValueError('Invalid compartment supplied: {}. '
+                         'Must be one of {}'.format(compartment, valid_compartments))
+
+    if compartment == 'whole-cell':
+        label_images = deep_watershed_mibi(model_output=model_output['whole-cell'],
+                                           **whole_cell_kwargs)
+    elif compartment == 'nuclear':
+        label_images = deep_watershed_mibi(model_output=model_output['nuclear'],
+                                           **nuclear_kwargs)
+    elif compartment == 'both':
+        label_images_cell = deep_watershed_mibi(model_output=model_output['whole-cell'],
+                                                **whole_cell_kwargs)
+
+        label_images_nucleus = deep_watershed_mibi(model_output=model_output['nuclear'],
+                                                   **nuclear_kwargs)
+
+        label_images = np.concatenate((label_images_cell, label_images_nucleus), axis=-1)
+
+    else:
+        raise ValueError('Invalid compartment supplied: {}. '
+                         'Must be one of {}'.format(compartment, valid_compartments))
+
+    return label_images
+
+
+def format_output_multiplex(output_list):
+    """Takes list of model outputs and formats into a dictionary for better readability
+
+    Args:
+        output_list: list of predictions from semantic heads
+
+    Returns:
+        dict: dictionary with predictions
+
+    Raises:
+        ValueError: model output list is not len(8)
+    """
+
+    if len(output_list) != 8:
+        raise ValueError('output_list was length {}, expecting length 8'.format(len(output_list)))
+
+    formatted_dict = {
+        'whole-cell': {
+            'inner-distance': output_list[0],
```

Comment:
This one got updated as well for 4 head model: https://github.com/vanvalenlab/deepcell-tf/blob/f1838ea700ea063e8f28c27da019e13526c43258/deepcell/applications/multiplex_segmentation.py#L69